tests/AstrolabeWorkloadExecutor/AstrolabeTestRunner.cs

```diff
@@ -0,0 +1,204 @@
+        {
+            SetupAndRunTest(testCase);
+        }
+
+        // protected methods
+        protected override void AssertOperation(JsonDrivenTest test)
+        {
+            var wrappedActualException = test._actualException();
+            if (test._expectedException() == null)
+            {
+                if (wrappedActualException != null)
+                {
+                    if (!(wrappedActualException is OperationCanceledException))
+                    {
+                        Console.WriteLine($"Operation error (unexpected exception type): {wrappedActualException.GetType()}");
+                        _incrementOperationErrors();
+                    }
+                    else
+                    {
+                        Console.WriteLine($"Operation cancelled: {wrappedActualException}");
+                    }
+
+                    return;
+                }
+                if (test._expectedResult() == null)
+                {
+                    _incrementOperationSuccesses();
+                }
+                else
+                {
+                    try
+                    {
+                        test.AssertResult();
+                        _incrementOperationSuccesses();
+                    }
+                    catch (Exception ex)
+                    {
+                        Console.WriteLine($"Operation failure (unexpected exception type): {ex.GetType()}");
+                        _incrementOperationFailures();
+                    }
+                }
+            }
+            else
+            {
+                if (wrappedActualException == null)
+                {
+                    _incrementOperationErrors();
+
+                    return;
+                }
+                try
+                {
+                    test.AssertException();
+                    _incrementOperationSuccesses();
+                }
+                catch
+                {
+                    _incrementOperationFailures();
+                }
+            }
+        }
+
+        protected override void RunTest(BsonDocument shared, BsonDocument test, EventCapturer eventCapturer)
+        {
+            Console.WriteLine("dotnet astrolabetestrunner> creating disposable client...");
+            using (var client = CreateClient(eventCapturer))
+            {
+                Console.WriteLine("dotnet astrolabetestrunner> looping until cancellation is requested...");
+                while (!_cancellationToken.IsCancellationRequested)
+                {
+                    // Clone because inserts will auto assign an id to the test case document
+                    ExecuteOperations(
+                        client: client,
+                        objectMap: new Dictionary<string, object>(),
+                        test: test.DeepClone().AsBsonDocument);
+                }
+            }
+
+            DisposableMongoClient CreateClient(EventCapturer eventCapturer)
```

Comment:
This should be a regular private method. It doesn't really make sense or help for this to be a local method.